src/libs/events.js

```diff
@@ -63,4 +65,48 @@ export const PageViewReporter = () => {
   return null
 }
 
+export const captureAppcuesEvent = (eventName, event) => {
+  // Only record "public-facing events" (and related properties) as documented by Appcues: https://docs.appcues.com/article/301-client-side-events-reference
+  const publicEvents = [
+    'flow_started',
+    'flow_completed',
+    'flow_skipped',
+    'flow_aborted',
+    'step_started',
+    'step_completed',
+    'step_skipped',
+    'step_aborted',
+    'step_interacted',
+    'form_submitted',
+    'form_field_submitted'
+  ]
+  if (_.includes(eventName, publicEvents)) {
+    const eventProps = {
```

Comment:
I think we can simplify this.

If we change the signature of this function to this:
```js
captureAppcuesEvent = (eventName, { interaction: { fields, ...interaction } = {}, ...event }) => {
```
We can build the result like this:

```js
const eventProps = _.flow(
  _.mapKeys(k => `appcues.${k}`),
  _.merge(_.mapKeys(k => `appcues.interaction.${k}`, interaction)),
  _.set(['appcues.interaction.fields'], fields && JSON.stringify(fields)), // note the [] here that let us set a key with a period in it
)(event)
```

That might need a bit of tweaking, but I think it should put you in the right direction.

Reply:
I considered trying something like this. I didn't want to take the time to write a recursive solution, and I'm not sure about the general transform with the special case for how to handle `interaction`. I also wanted to reduce any risk of this code breaking due to a change in the Appcues event structure, and for it to be easy to fix if it ever breaks. It is a bit verbose, but I think it's simple and easy to understand.